File: backend/app/models/tool.rb
class Tool < ApplicationRecord
  CATEGORIES = %w[Hand Power Clamps Measuring Sharpening Safety Fastening Finishing Other].freeze

  validates :name, presence: true
  validates :quantity, numericality: { only_integer: true, greater_than_or_equal_to: 0 }

  scope :ordered, -> { order(:category, :name) }
end

File: frontend/src/app/tool.ts
export interface Tool {
  id?: number;
  name: string;
  category: string | null;
  brand: string | null;
  model: string | null;
  quantity: number;
  location: string | null;
  /** Photo of the tool, stored as a downscaled JPEG data URL (see image.ts). */
  image?: string | null;
  created_at?: string;
  updated_at?: string;
}

export const TOOL_CATEGORIES = [
  'Hand',
  'Power',
  'Clamps',
  'Measuring',
  'Sharpening',
  'Safety',
  'Fastening',
  'Finishing',
  'Other'
] as const;

export function emptyTool(): Tool {
  return { name: '', category: '', brand: '', model: '', quantity: 1, location: '', image: null };
}
